pkg/apiserver/experiment/experiment.go

```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
 // TODO: need to be implemented
 func (s *Service) deleteExperiment(c *gin.Context) {}
 
+// ExperimentInfo defines a form data of Experiment from API.
+type ExperimentInfo struct {
+	Name      string        `json:"name" binding:"required,NameValid"`
+	Namespace string        `json:"namespace" binding:"required,NameValid"`
+	Scope     ScopeInfo     `json:"scope"`
+	Target    TargetInfo    `json:"target"`
+	Scheduler SchedulerInfo `json:"scheduler"`
+}
+
+// ScopeInfo defines the scope of the Experiment.
+type ScopeInfo struct {
+	NamespaceSelectors  []string          `json:"namespace_selectors" binding:"NamespaceSelectorsValid"`
+	LabelSelectors      map[string]string `json:"label_selectors" binding:"MapSelectorsValid"`
+	AnnotationSelectors map[string]string `json:"annotation_selectors" binding:"MapSelectorsValid"`
+	FieldSelectors      map[string]string `json:"field_selectors" binding:"MapSelectorsValid"`
+	PhaseSelector       []string          `json:"phase_selectors" binding:"PhaseSelectorsValid"`
+
+	Mode string `json:"mode" binding:"required,oneof=one all fixed fixed-percent random-max-percent"`
+}
+
+func (s *ScopeInfo) parseSelector() v1alpha1.SelectorSpec {
+	selector := v1alpha1.SelectorSpec{}
+
+	for _, ns := range s.NamespaceSelectors {
+		selector.Namespaces = append(selector.Namespaces, ns)
+	}
+
+	selector.LabelSelectors = make(map[string]string)
+	for key, val := range s.LabelSelectors {
+		selector.LabelSelectors[key] = val
+	}
+
+	selector.AnnotationSelectors = make(map[string]string)
+	for key, val := range s.AnnotationSelectors {
+		selector.AnnotationSelectors[key] = val
+	}
+
+	selector.FieldSelectors = make(map[string]string)
+	for key, val := range s.FieldSelectors {
+		selector.FieldSelectors[key] = val
+	}
+
+	for _, ph := range s.PhaseSelector {
+		selector.PodPhaseSelectors = append(selector.PodPhaseSelectors, ph)
+	}
+
+	return selector
+}
+
+// TargetInfo defines the information of target objects.
+type TargetInfo struct {
+	Kind         string       `json:"kind" binding:"required,oneof=PodChaos NetworkChaos IOChaos KernelChaos TimeChaos StressChaos"`
+	PodChaos     PodChaosInfo `json:"pod_chaos"`
+	NetworkChaos NetworkChaosInfo
+	IOChaos      IOChaosInfo
+	KernelChaos  KernelChaosInfo
+	TimeChaos    TimeChaosInfo
+	StressChaos  StressChaosInfo
+}
+
+// SchedulerInfo defines the scheduler information.
+type SchedulerInfo struct {
+	Cron string `json:"cron" binding:"CronValid"`
+}
```

Comment:
Where is the `Duration`? I guess people may use `duration` for `pod-failure` chaos.